src/hotspot/cpu/x86/x86.ad

```diff
@@ -4834,6 +4847,51 @@ instruct reduction16F(regF dst, legVec src, legVec vtmp1, legVec vtmp2) %{
   ins_pipe( pipe_slow );
 %}
 
+// =======================Half Float Reduction==========================================
+instruct reduction8HF(rRegI dst, vec src2, vec tmp, vec tmp1, vec tmp2) %{
+  predicate(UseAVX > 2);
```

Comment:
This patterns handles 128 bit vector, in case we defer handling 256/512 bit vectors we should add a check in match_rule_supported_vector.